src/Sarif.Viewer.VisualStudio.Core/Tags/SarifLocationErrorTag.cs

```diff
@@ -42,6 +72,101 @@ public SarifLocationErrorTag(IPersistentSpan persistentSpan, int runIndex, int r
         /// <remarks>
         /// This may be null.
         /// </remarks>
-        public object ToolTipContent { get; }
+        private readonly List<(string strContent, TextRenderType renderType)> content;
+
+        public object ToolTipContent
+        {
+            get
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+                foreach ((string strContent, TextRenderType renderType) item in content)
+                {
+                    if (item.renderType == TextRenderType.Markdown)
+                    {
+                        try
+                        {
+                            MarkdownViewer viewer = new MarkdownViewer();
+                            viewer.Markdown = item.strContent;
+                            ParseBlocks(viewer.Document.Blocks);
+
+                            viewer.Margin = new Thickness(-15, -15, 0, 0); // There is a small amount of padding that MarkdownViewer comes with that makes it awkward when a textfield is put alongside it.
+                            return viewer;
+                        }
+                        catch (Exception)
+                        {
+                            // catch and swallow silently
+                        }
+                    }
+                    else if (item.renderType == TextRenderType.Text)
+                    {
+                        TextBlock textblock = new TextBlock() { Text = item.strContent };
+                        textblock.FontSize = fontSize;
+                        textblock.Foreground = textBrush;
+                        return textblock;
+                    }
+                    else
+                    {
+                        throw new ArgumentException($"Unknown text render type {item.renderType}");
```

Comment:
>throw new ArgumentException($"Unknown text render type {item.renderType}"); [](http://example.com/codeflow?start=24&length=75)

I think we should simply return null in this case: this is not exceptional and doesn't leave the app in a bad state.

Reply:
Just curious what do you consider to be the line between something that is exceptional and something that is not? 
When writing this my logic was similar to [here](https://github.com/microsoft/sarif-visualstudio-extension/pull/589#discussion_r1130015372) where I want to raise exceptions on unexpected situations to catch them during testing rather than let them pass silently

Reply:
Since the property is an enum, it really can't be unknown. If you want to guard against new enum values being added, then I guess throwing wouldn't be outrageous. Maybe change the error to "Unsupported text render type" and list the ones that are supported.